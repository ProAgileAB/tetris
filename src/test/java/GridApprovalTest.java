import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;


class GridApprovalTest {

    // # initial state - an empty grid
    // # a grid with top left cell set to letter A
    // # a grid bottom right cell set to number
    // A grid with Yellow Block in center

    @Test
    void anEmptyGrid() {
        var grid = new MyBoundedGrid<String>(2, 3);
        Approvals.verify(grid);
    }

    @Test
    void aGridWithTopLeftCellsSetToA() {
        var grid = new MyBoundedGrid<String>(2, 3);
        grid.put(new Location(0, 0), "A");
        Approvals.verify(grid);
    }

    @Test
    void aGridWithBottomRightCellsSetTo1() {
        var grid = new MyBoundedGrid<Integer>(2, 3);
        grid.put(new Location(1, 2), 1);
        Approvals.verify(grid);
    }

    @Test
    void aGridWithBlueBlockInCenter() {
        var grid = new MyBoundedGrid<Block>(3, 3);
        grid.put(new Location(1, 1), new Block());
        Approvals.verify(grid);
    }
}