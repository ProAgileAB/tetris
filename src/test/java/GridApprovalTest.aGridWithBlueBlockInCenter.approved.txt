A grid of 3 rows 3 columns. Content:
...
.B.
...